docs/input/docs/usage/command-line.md

```diff
@@ -119,14 +119,65 @@ Will iterate through each file and update known attributes (`AssemblyVersion`,
 
 `/overrideconfig [key=value]` will override appropriate key from 'GitVersion.yml'.
 
-At the moment only `tag-prefix` option is supported. Read more about [Configuration](../configuration).
+Multiple options are separated by semicolon: `key1=value1;key2=value2`.
```

Comment:
This is well-known aproach used by MSBuild command line, to accept multiple key-value pairs with `;`. For example:

> msbuild /target:build /property:Configuration=Release;TargetFramework=net5.0

Multiple `/property` entries are possible in MSBuild as well. So since your current command-line syntax reassembles MSBuild syntax quite and a lot of developers will use it in the same build scripts, I thought it's good idea to follow already established syntax.

Regarding your concrete suggestion, space character cannot work as a separator for key-values, because key would be confused for next command line switch.

Adding multiple occurrences of `/overrideconfig` argument would be possible, there is an issue with current implementation as well.

My parsing of additional overrideconfig keys hooked into existing implementation, at the point where command line switches are already parsed. Namely inside of the method: `ParseOverrideConfig` of `ArgumentParser`. As far as I can tell, accepting multiple occurrences of `/overrideconfig` would require rewrite of complete command line parsing logic. Perhaps someone who is familiar with current implementation can give us 2nd opinion on this change?

Reply:
> This is well-known aproach used by MSBuild command line, to accept multiple key-value pairs with `;`

Yea, I think MSBuild is a horrendous example of command line interface and GitVersion should not be modelled by it. The more different from MSBuild we can make GtiVersion's CLI, the better. 😄 

> Multiple `/property` entries are possible in MSBuild as well.

Good. 👍🏼 

> Regarding your concrete suggestion, space character cannot work as a separator for key-values, because key would be confused for next command line switch.

Keys and values should be separated with `=`. But key-value-pairs should be separated by space.

> Perhaps someone who is familiar with current implementation can give us 2nd opinion on this change?

If it's too involved, let's postpone this to GitVersion 6 where we will redo the entire CLI (see #2275).

Reply:
> Yea, I think MSBuild is a horrendous example of command line interface and GitVersion should not be modelled by it. The more different from MSBuild we can make GtiVersion's CLI, the better. 😄

Wherever I found command line switch that accept multiple values, they are splitted with `;` or `,` or some other funky character. For example `:` in `--resolve`:

`curl --resolve 'yada.com:443:127.0.0.1' https://yada.com/something`

> Keys and values should be separated with =. But key-value-pairs should be separated by space.

That's what I was referring too. Normally, arguments are split by space characters and we get array:

`switch value switch value switch value (etc)`

If we split key-value pairs with spaces, we suddenly get ambiguity, which argument is command line switch, which one is value of command line switch, which one is key value pair.

`switch value1 value2 switch value switch value`

It creates complexities for parser. Add on top of that that some switches might repeat and it becomes a mess. And I also think, maybe it's on the surface prettier, but it quickly becomes less human-readable to figure what is switch and what is key-value pair in a long command.

> If it's too involved, let's postpone this to GitVersion 6 where we will redo the entire CLI

Let me explain why I think it's this is a big change. Currently, repeated parameter `/output` will just be ignored:

`gitversion /output json /output buildserver`

In implementation where multiple occurrences are valid, you will have 2 cases now. One case where, like `/output` one value of parameter overrides another value. And second case where you want to merge key-value pairs from multiple occurrences. If key-value appear multiple times, then they should be overwritten/thrown away.

Regardless where this option is introduced, in GitVersion 5 or 6, it will introduce additional complexity that is currently not required.

Reply:
> `switch value switch value switch value (etc)`

No, I was thinking like this:

```shell
gitversion /overrideconfig key1=value1 key2=value2 key3=value3
```

Each key-value pair is separated with space, key and value is separated by `=`.

Reply:
Yes, that we are talking about the same thing. This is 2nd example I made:

`switch value1 value2 switch value switch value`

Or translated to your example:
`gitversion /overrideconfig key1=value1 key2=value2 /ouput json /output buildserver`

The way how arguments are parsed universally, it's the array: `/overrideconfig`, `key1=value`, `key2=value`, `/output`, etc... Further, order is assumed/convention, option is followed by the value. Following that logic, `key2=value` ends up in a position of command line option and `/output` in the place of value.

What you are proposing introduces variable positions of options, values and child-values. And that introduces all kinds of combinatorial issues and need for top level parser to know all valid values of child-options to do parse this correctly You end up with mixing different concerns on the top level.

Reply:
I see. It's not correct that the top-level parser needs to know what the underlying switches or values are, it just needs to know the difference between switches (arguments prefixed by `/`, `-` or `--`) and values (one or more arguments following a switch).

The way to implement this is to first split by space, then loop through each argument and stack switches with values before the values are passed to the switch handler. In the case of `/overrideconfig`, the values passed would be the array `["key1=value1", "key2=value2", "key3=value3]` and then it would be up to the switch handler to split the keys and values.

 understand that the current implementation is too naïve to take this into consideration. I still want to investigate alternatives to semicolon, though. Ideas?